stdlib/weak.ml

```diff
@@ -238,8 +238,11 @@ module Make (H : Hashtbl.HashedType) : (S with type data = H.t) = struct
     let bucket = t.table.(index) in
     let hashes = t.hashes.(index) in
     let sz = length bucket in
-    let rec loop i =
-      if i >= sz then begin
+    let i = ref 0 in
+    let continue = ref true in
+    while !continue do
```

Comment:
Instead of using a ref, the body of the loop could be moved inside the condition expression
```
while
  <Boby>
do () done
```
where body evaluate to true if the loop need to continue.

Reply:
Yes, I like this pattern too but it horrifies some people so I tend not to use it in public code. If there is support for this from my fellow maintainers I will gladly switch to this version.

Reply:
I guess you wanted to minize the diff. But wouldn't an idiomatic loop rather look like:
```ocaml
while !i < sz && check_bucket !i do incr i done 
if !i = sz then … else …
```

Reply:
Good point. I will admit that I didn't even check what the loop was doing; I will amend with your suggestion.

Reply:
(I also assumed that you went for the minimal change, but I'm also okay with a more readable new version.)